src/Model/Actions.re

```diff
@@ -18,7 +18,7 @@ module Diagnostic = Feature_LanguageSupport.Diagnostic;
 
 [@deriving show({with_path: false})]
 type t =
-  | Init
+  | Init([@opaque] Cli.t)
```

Comment:
Why not just pass this as a parameter to `State.create` instead?  The would eliminate the need to initialize the state with a default that might not make sense, and might even introduce subtle timing bugs.

It should also be more explicit. Don't pass the entire Cli object when all you need is a single value from it. Doing so obscures the actual dependency and adds an unnecessary dependency on the Cli object. Feature stores shouldn't need to know about the existence of a CLI interface.

Reply:
> Why not just pass this as a parameter to State.create instead? The would eliminate the need to initialize the state with a default that might not make sense, and might even introduce subtle timing bugs.

I tried this and it had its own set of issues, because of the way we use state in `GlobalContext` right now. We'd need to plumb the Cli.t through the singleton, so I'm not sure it's much better. Removing the `GlobalContext` would help this...

But maybe I'm missing something. Feel free to push up a change if you see a cleaner way to do this!

> Don't pass the entire Cli object when all you need is a single value from

In the meantime, I went this route, and just simplified the `Cli.t` to only have a syntaxHighlightEnabled parameter.

Reply:
> Removing the GlobalContext would help this...

Ah, yeah that's a wart I'd like to get rid of too. Having this on `Init` isn't that bad until we can get rid of it. We'll want to eventually get rid of `Init` entirely too I think, so can clean it up then.